src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -519,12 +514,15 @@ public Person generatePerson(int index, long personSeed) {
   /**
    * Determines if a patient meets the requested criteria.
    * If a patient does not meet the criteria the process will be repeated so a new one is generated
+   * @param person the patient to check if we want to export them
+   * @param finishTime the time simulation finished
    * @param isAlive Whether the patient is alive at end of simulation.
    * @param providerCount Number of providers in the patient's record
    * @param providerMinimum Minimum number of providers required
    * @return true if patient meets criteria, false otherwise
    */
-  public boolean patientMeetsCriteria(boolean isAlive, int providerCount, int providerMinimum) {
+  public boolean patientMeetsCriteria(Person person, long finishTime, boolean isAlive,
```

Comment:
This is fine and maybe I think too much in OO, but I'm wondering if we can clean things up by introducing a class for the result of a patient simulation. Call it `SimulationResult` and it can have the person, finish time, whether they are alive and the provider count.

The `patientMeetsCriteria` could be moved to that class and get passed the options for the overall simulation or work with `Config`. Right now, things like `onlyDeadPatients` get set as fields on `Generator`, but are really only used in `patientMeetsCriteria`, outside of checking that `onlyDeadPatients` and `onlyAlivePatients` conflict.

Reply:
Where this gets tricky is in all the various methods that return a Person object. And in `Person generatePerson(int index, long personSeed)` it's extra tricky because that function generates many but only returns one. 
I think you're probably right, we can clean things up (do we really need all those different functions?) but the delta to do that will be nontrivial.
I've made a couple additional tweaks to reduce the number of variables that get passed around, so hopefully it seems a little cleaner now.